### ISSUE TITLE
PlayStore crash: cast exception

### ISSUE THREAD

**Issue by andyboeh** [sn-00-x/fakestore2playstore#6]
I had this module installed for a few monhts (thanks for that!), since yesterday, PlayStore keeps crashing with a cast exception as soon as I interact with it (it opens just fine).

I'm on LOS 4 microg, tested on two different devices. Deleting data/cache doesn't help.

Edit: I suppose that's the reason why microg_installer_revived switched to the unpatched version of PS. Any plans to update this?

**Comment by erdemozdemir98** [comment]
I finally found a solution to this issue.

Prerequisites:
>You need to be rooted with Magisk.
>Your SafetyNet should pass (not exactly sure about that).
>Fake Store or Play Store must be installed as a system app.
>If your data is encrypted you probably need to use a bootloop protector module.
>You need to have signature spoofing enabled in your ROM (or you can use the FakeGapps module with LSPosed).

Steps you need to do:
>Install Lucky Patcher, grant all permissions (including root).
>In Lucky Patcher, go to toolbox and select patch Android. Enable using the Magisk module, enable first three patches, apply and reboot.
>Install the latest vanilla Play Store apk from apkmirror. Grant all permissions.
>Play Store should be working flawlessly now.

Possible bugs:
>Play Store works significantly slower than the patched one but it gets the work done. This also might be a device problem, I'm not sure about that. Plus, vanilla Play Store's certification system works better for me compared to the ancient patched Play Store apk.
>You might need to hide Magisk from Play Store. You can use Magisk Delta's built-in Hide functionality for that.
>License verification randomly stops working, but if you try again it's almost guaranteed to work. This might be a device problem too.
>If Play Store still keeps crashing, try force stopping it and deleting the cache.

P.S.: I tried MinMicroG and MicroG Installer Revived. Their Play Store crashed for me, so I tried this module. It still crashed. I also tried countless possible solutions but only this method worked for me. I hope it works for you too.

**Comment by bege10** [comment]
A few days ago the Play Store crashed here also when trying to search for apps or list the installed apps.
The next day it worked again. Today it crashes again as it does on another phone with a fresh installed iodéOS.
I won't use Lucky Patcher, so I hope for another solution.

**Comment by erdemozdemir98** [comment]
> A few days ago the Play Store crashed here also when trying to search for apps or list the installed apps. The next day it worked again. Today it crashes again as it does on another phone with a fresh installed iodéOS. I won't use Lucky Patcher, so I hope for another solution.

You can uninstall Lucky Patcher and all things related to it once the patching is done. Also the only usage of Lucky Patcher here is to patch the Android system for allowing different signatures on the same package. If you know another tool or method for that, you can use it too.

**Comment by bege10** [comment]
> to patch the Android system for allowing different signatures on the same package

Do you still use LineageOS for microG or is your workaround without microG?

**Comment by erdemozdemir98** [comment]
> > to patch the Android system for allowing different signatures on the same package
> 
> Do you still use LineageOS for microG or is your workaround without microG?

I use LineageOS with FakeGApps and microG installed on top of it. I guess my workaround won't work without microG because it requires Fake Store or (patched) Play Store as a system app in the first place.

**Comment by bege10** [comment]
I solved it with the Vending app and permissions (Phonesky does not install correctly) from this module
https://github.com/Magisk-Modules-Alt-Repo/microG-GApps

Edit: Some time later, another fresh installation of the ROM and the same procedure does not work any more. 🤔

**Comment by D3SOX** [comment]
@bege10 How did you install it without the microG part? I tried to replace the permissions and APK file from this module ZIP and install it, but it didn't work.

**Comment by bege10** [comment]
> @bege10 How did you install it without the microG part? I tried to replace the permissions and APK file from this module ZIP and install it, but it didn't work.

It needs microG. I used /e/OS and LineageOS for microG, both have microG already included.

**Comment by D3SOX** [comment]
Yeah, but I think I previously broke my Lineage for microG installation by installing the whole module

**Comment by bege10** [comment]
> Yeah, but I think I previously broke my Lineage for microG installation by installing the whole module

That should be fixed by simply deactivating or removing the module.
